client/components/data/query-dotorg-plugins/index.jsx

```diff
@@ -0,0 +1,42 @@
+import config from '@automattic/calypso-config';
+import PropTypes from 'prop-types';
+import { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useInterval } from 'calypso/lib/interval';
+import { fetchPluginData as wporgFetchPluginData } from 'calypso/state/plugins/wporg/actions';
+import { getAllPlugins } from 'calypso/state/plugins/wporg/selectors';
+
+const PLUGIN_RETRIEVE_INTERVAL = 200;
+
+function QueryDotorgPlugins( { pluginSlugList } ) {
+	const dispatch = useDispatch();
+	const queueRef = useRef( [] );
+
+	const dotorgPlugins = useSelector( ( state ) => getAllPlugins( state ) );
+
+	useEffect( () => {
+		pluginSlugList.forEach( ( pluginSlug ) => {
+			if ( ! queueRef.current.includes( pluginSlug ) && ! dotorgPlugins[ pluginSlug ] ) {
+				queueRef.current.push( pluginSlug );
+			}
+		} );
+	}, [ pluginSlugList, dotorgPlugins ] );
+
+	useInterval( () => {
+		if ( pluginSlugList.length ) {
+			const pluginSlug = queueRef.current.shift();
+
+			if ( pluginSlug && config.isEnabled( 'bulk-plugin-management' ) ) {
```

Comment:
maybe we shouldn't start any processing at all if feature not enabled. maybe move this up below the selector and return immediately if not enabled?

Reply:
> maybe move this up below the selector 

I'd love to, but since `useInterval` uses the `useEffect` selector inside it, we can't add a condition before it.